mantidproject/mantid
# Make sanitizers work with Python

**Description of work.**

This is the final missing piece of the puzzle, I couldn't figure out why
LD_PRELOAD was not working. It turns out ctest calls bin/mantidpython,
which has it's own override for LD_PRELOAD

With this we can run sanitizers for both C++ and Python natively without
messing around with scripts. Hopefully this work will enable us to flush
out some flakey tests.

I wasn't intending to do this work today, but I stumbled upon the fix whilst trying to get the sanitizers working locally, so it made sense to package it before I forgot it.

**To Test**
- Run `ctest -R Kernel -VV` check `ASAN_OPTIONS` and `LSAN_OPTIONS` are both there
- Run `ctest -R python -VV` check that LD_PRELOAD errors aren't emitted (you will likely see leaks)

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
